Can't resolve `graphql-ws` from `@graphiql/toolkit/esm/create-fetcher`

Seems like the recent release of `@graphiql/toolkit` has broken some of our tests. Maybe due to a missing dependency?

```
Module not found: Error: Can't resolve 'graphql-ws' in '/tmp/backstage-e2e-Z2eQgF/test-app/node_modules/@graphiql/toolkit/esm/create-fetcher'

error Command failed with exit code 1.
```

From https://github.com/backstage/backstage/runs/6420680730?check_suite_focus=true

--- a/.changeset/four-mangos-warn.md
+++ b/.changeset/four-mangos-warn.md
@@ -1,0 +1,5 @@
+---
+'@graphiql/toolkit': patch
+---
+
+Move `graphql-ws` dependency to a dynamic import, and add a nice error message when it's not installed

--- a/packages/graphiql-toolkit/src/create-fetcher/__tests__/lib.spec.ts
+++ b/packages/graphiql-toolkit/src/create-fetcher/__tests__/lib.spec.ts
@@ -102,3 +102,18 @@ describe('getWsFetcher', () => {
     expect(SubscriptionClient.mock.calls).toHaveLength(0);
   });
 });
+
+describe('missing graphql-ws dependency', () => {
+  it('should throw a nice error', () => {
+    jest.resetModules();
+    jest.doMock('graphql-ws', () => {
+      throw { code: 'MODULE_NOT_FOUND' };
+    });
+
+    expect(() =>
+      createWebsocketsFetcherFromUrl('wss://example.com'),
+    ).toThrowError(
+      /You need to install the 'graphql-ws' package to use websockets when passing a 'subscriptionUrl'/,
+    );
+  });
+});

--- a/packages/graphiql-toolkit/src/create-fetcher/lib.ts
+++ b/packages/graphiql-toolkit/src/create-fetcher/lib.ts
@@ -1,10 +1,10 @@
 import { DocumentNode, visit, GraphQLError } from 'graphql';
 import { meros } from 'meros';
 import {
-  createClient,
   Client,
   ClientOptions,
   ExecutionResult,
+  createClient as createClientType,
 } from 'graphql-ws';
 import {
   isAsyncIterable,
@@ -18,6 +18,10 @@ import type {
   ExecutionResultPayload,
   CreateFetcherOptions,
 } from './types';
+
+const errorHasCode = (err: unknown): err is { code: string } => {
+  return typeof err === 'object' && err !== null && 'code' in err;
+};
 
 /**
  * Returns true if the name matches a subscription in the AST
@@ -76,6 +80,10 @@ export const createWebsocketsFetcherFromUrl = (
 ) => {
   let wsClient;
   try {
+    const { createClient } = require('graphql-ws') as {
+      createClient: typeof createClientType;
+    };
+
     // TODO: defaults?
     wsClient = createClient({
       url,
@@ -83,6 +91,13 @@ export const createWebsocketsFetcherFromUrl = (
     });
     return createWebsocketsFetcherFromClient(wsClient);
   } catch (err) {
+    if (errorHasCode(err)) {
+      if (err.code === 'MODULE_NOT_FOUND') {
+        throw Error(
+          "You need to install the 'graphql-ws' package to use websockets when passing a 'subscriptionUrl'",
+        );
+      }
+    }
     console.error(`Error creating websocket client for:\n${url}\n\n${err}`);
   }
 };

